flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyedStateBackend.java

```diff
@@ -136,6 +138,15 @@
      */
     boolean deregisterKeySelectionListener(KeySelectionListener<K> listener);
 
+    /** Returns the total number of state entries across all keys/namespaces. */
+    @VisibleForTesting
+    int numKeyValueStateEntries();
```

Comment:
I'd like to avoid adding methods exclusively for testing to public production interfaces.
Instead, it can be in `AbstractKeyedStateBackend`.

Reply:
That's a very good point, thanks!

Reply:
I remembered I moved it from `AbstractKeyedStateBackend` to `KeyedStateBackend.java`.